### PR TITLE
添加 Parser 与 Builder | Add Parser and Builder

### PR DESCRIPTION
**Pull request by wfnuser**
data parser 翻译

### CODE REVIEW

File: Z-Data/Parser-and-Builder.md [hunk 1]
```diff
@@ -1,19 +1,26 @@
 ---
 layout: default
 parent: Z-Data
-title: Parser and Builder
+title: 解析器(Parser)和创建器(Builder)
```
Comment (alissa-tung) [inline Z-Data/Parser-and-Builder.md]: I recommend that use titles of form `<org>：<zh>`. Parser and Builder are (verb.)r so maybe there is another to make them more precise. How does textbooks in Chinese called them？

File: Z-Data/Parser-and-Builder.md [hunk 2]
```diff
@@ -72,25 +86,45 @@ msgStr = do
             Just t -> return (Str t)
             _  -> P.fail' "illegal UTF8 Bytes"
 ```
-
+{::comment}
 Comparing to `parsec` or `megaparsec`, `Parser` in Z provides limited error reporting, and do not support using as a monad transformer. But provides an instance of `PrimMonad`, which allows some limited effects, such as mutable variables and array operations. 
+{:/}
 
+与 `parsec` 或 `megaparsec` 相比，Z中的 `Parser` 提供有限的错误报告，并且不支持将其用作 monad 转换器。 但是提供了 `PrimMonad` 的实例，该实例允许一些有限的副作用，例如可变变量和数组操作。
```
Comment (alissa-tung) [inline Z-Data/Parser-and-Builder.md]: 与…不同；可变变量感觉有点怪怪的？可以讨论下这种，感觉是常见的范式

File: Z-Data/Parser-and-Builder.md [hunk 3]
```diff
@@ -72,25 +86,45 @@ msgStr = do
...
+
+在上面的代码中，如果任何 parser 失败，则从输入的开头重试下一个 parser。虽然并非总是需要回溯，还是建议使用 `peek`。或者在语法或协议可以被解析为 LL(1) 语法的时候使用`peekMaybe`， 因为它比回溯更快。
 
 # Builder Monad
 
+{::comment}
 The `Builder` from `Z.Data.Builder` is the reverse process of parsing, i.e. writing Haskell data types to `Bytes`, aka *Writer* monad. The usage is very similiar to `Parser`:
+{:/}
+
+`Z.Data.Builder` 中的 `Builder` 是解析过程的逆过程，即将Haskell数据类型写入 `Bytes` ，也就是 *Writer* monad。用法非常类似于 `Parser` ：
```
Comment (alissa-tung) [inline Z-Data/Parser-and-Builder.md]: 差个空格

**Review by alissa-tung — CHANGES_REQUESTED** [review]

File: Z-Data/Parser-and-Builder.md (hunk 1 above)
Comment (wfnuser) [reply Z-Data/Parser-and-Builder.md]: I guess I will just keep the words `Parser` and `Builder` untranslated then.
But... I think Parser and Builder are actually (noun.) ?

File: Z-Data/Parser-and-Builder.md (hunk 3 above)
Comment (wfnuser) [reply Z-Data/Parser-and-Builder.md]: done

File: Z-Data/Parser-and-Builder.md (hunk 1 above)
Comment (alissa-tung) [reply Z-Data/Parser-and-Builder.md]: 我打算说解析和创建在中文语境下都是依赖上下文的 0.0 我之前的想法是不翻译，反正用起来的时候 import 和 typing 都是原词

**Pull request merged**